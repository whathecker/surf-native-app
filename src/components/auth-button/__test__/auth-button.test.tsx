import * as React from "react";
import renderer from "react-test-renderer";
import AuthButton from "../auth-button";
//import { AuthButtonProps } from '../types';

describe("Test AuthButton component", () => {
  it("should renders apple auth button correctly for sign-up screen", () => {
    const tree = renderer.create(<AuthButton screenType="signup" authBrand="apple" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it("should renders facebook auth button correctly for sign-up screen", () => {
    const tree = renderer.create(<AuthButton screenType="signup" authBrand="facebook" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it("should renders google auth button correctly for sign-up screen", () => {
    const tree = renderer.create(<AuthButton screenType="signup" authBrand="facebook" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should renders apple auth button correctly for sign-in screen", () => {
    const tree = renderer.create(<AuthButton screenType="signin" authBrand="apple" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it("should renders facebook auth button correctly for sign-in screen", () => {
    const tree = renderer.create(<AuthButton screenType="signin" authBrand="facebook" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it("should renders google auth button correctly for sign-in screen", () => {
    const tree = renderer.create(<AuthButton screenType="signin" authBrand="facebook" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
