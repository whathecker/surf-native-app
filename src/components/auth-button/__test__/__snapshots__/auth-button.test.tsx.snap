// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AuthButton component should renders apple auth button correctly for sign-in screen 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "black",
      "borderRadius": 12,
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "opacity": 1,
      "paddingBottom": 18,
      "paddingTop": 18,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "marginRight": 35,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 30,
          "marginLeft": 20,
          "width": 30,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 16,
      }
    }
  >
    Continue with Apple
  </Text>
</View>
`;

exports[`Test AuthButton component should renders apple auth button correctly for sign-up screen 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "black",
      "borderRadius": 12,
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "opacity": 1,
      "paddingBottom": 18,
      "paddingTop": 18,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "marginRight": 35,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 30,
          "marginLeft": 20,
          "width": 30,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 16,
      }
    }
  >
    Sign up with Apple
  </Text>
</View>
`;

exports[`Test AuthButton component should renders facebook auth button correctly for sign-in screen 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "blue",
      "borderRadius": 12,
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "opacity": 1,
      "paddingBottom": 18,
      "paddingTop": 18,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "marginRight": 35,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 30,
          "marginLeft": 20,
          "width": 30,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 16,
      }
    }
  >
    Continue with Facebook
  </Text>
</View>
`;

exports[`Test AuthButton component should renders facebook auth button correctly for sign-up screen 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "blue",
      "borderRadius": 12,
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "opacity": 1,
      "paddingBottom": 18,
      "paddingTop": 18,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "marginRight": 35,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 30,
          "marginLeft": 20,
          "width": 30,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 16,
      }
    }
  >
    Sign up with Facebook
  </Text>
</View>
`;

exports[`Test AuthButton component should renders google auth button correctly for sign-in screen 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "blue",
      "borderRadius": 12,
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "opacity": 1,
      "paddingBottom": 18,
      "paddingTop": 18,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "marginRight": 35,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 30,
          "marginLeft": 20,
          "width": 30,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 16,
      }
    }
  >
    Continue with Facebook
  </Text>
</View>
`;

exports[`Test AuthButton component should renders google auth button correctly for sign-up screen 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "blue",
      "borderRadius": 12,
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "opacity": 1,
      "paddingBottom": 18,
      "paddingTop": 18,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "marginRight": 35,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 30,
          "marginLeft": 20,
          "width": 30,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 16,
      }
    }
  >
    Sign up with Facebook
  </Text>
</View>
`;
